Restore test cases

ddur opened the pull request:
Restore test cases

sawilde commented:
I don't understad the purpose of these tests that .dll, a process cannot be a .dll can it?

ddur commented:
It is to confirm match & mismatch of start and of end of process-name, simple string filter matching test.

But yes, I could use never matching ".dll" extension in **filter** instead giving it to **process name**, that would be the same logic. 

BTW, in later commits, I added more and removed duplicate tests.
https://github.com/OpenCover/opencover/pull/454
https://github.com/OpenCover/opencover/pull/461
https://github.com/OpenCover/opencover/pull/462

ddur commented:
It is true that using .dll extension could be confusing.
.dll extension moved from process name to filter
https://github.com/OpenCover/opencover/commit/2c7ba27c9c3c3f5f59742fd67b7231171f3945b8
